baseplate/lib/events.py

```diff
@@ -93,10 +95,18 @@ class EventQueue(ContextFactory, config.Parser, Generic[T]):
 
     """
 
-    def __init__(self, name: str, event_serializer: Callable[[T], bytes]):
-        self.queue = MessageQueue(
-            "/events-" + name, max_messages=MAX_QUEUE_SIZE, max_message_size=MAX_EVENT_SIZE
-        )
+    def __init__(
+        self,
+        name: str,
+        event_serializer: Callable[[T], bytes],
+        queue_type: QueueType = QueueType.POSIX,
+    ):
+        if queue_type == QueueType.IN_MEMORY:
+            self.queue = RemoteMessageQueue("/events-" + name, max_messages=MAX_QUEUE_SIZE)
+        else:
```

Comment:
Again you have set a default in the init and then encoded one in this if statement. A QueueType to queue implementation helper function could be used here too and then you don't have to worry if the default queue type changes in funture.